### DISCUSSION TITLE
xarray DataArray output -- grouping by spatial bins

### DISCUSSION THREAD

**Discussion by frazermi268** [miniscope/minian#203] [Q&A]
Hi there,

I just started using minian recently, and am having a bit of trouble understanding the structure of the dataset. I was wondering if anyone had advice for working with the minian data set output in 2D space. Specifically, I am trying to find place cells in an open field, and I want to bin the calcium event data (S) into ~2.5 cm square bins that cover the area. While doing this with a linear track makes sense, through using the grouby_bins functionality, I haven't been able to find a way to groupby_bins across both X and Y values, and it seems like the groupby functions for xr.DataSets only work for 1 dimension at a time.

Does anyone know whether there is a workaround for this? Or is this something that requires the data to be moved outside of the xr.DataArray format?

Thank you so much in advance!!

Michelle

**Comment by phildong** [comment]
This indeed seem to be outside of the use case for `groupby_bins`. However I believe `groupby_bins` is using `pandas.cut` under the hood and you can use that for your specific 2d case. Assuming you already assigned `x` and `y` coordinates to `S` you can try something like this:
```python
import itertools as itt
agg_ls = []
for binx, biny in itt.product(pd.cut(S.coords['x']), pd.cut(S.coords['y'])):
    agg_ls = S.sel(x=slice(*binx), y=slice(*biny))
```
